Review terraform (#1938)
newrelic/developer-website: src/markdown-pages/terraform/terragrunt-configuration.mdx

[output] "terraform apply" is subsequently run.
```

Terragrunt provides the `inputs` block's values.

</Step>

<Step>

Run `terragrunt apply`:

```bash
terragrunt apply
```

Now your Demo Policy is in your New Relic account.

</Step>

</Steps>

## Add to your configuration

Now that you've created a basic New Relic configuration, add the configurations from our [Getting Started with Terraform](/terraform/get-started-terraform) and [Terraform modules](/terraform/terraform-modules) guides.

<Callout variant="tip">

If you haven't done these guides yet, you can copy their configurations from the [Terragrunt intro Github repo](https://github.com/jsbnr/nr-terragrunt-intro-example).

</Callout>

<Steps>

<Step>

> In _main.tf_, update the email address in the alert channel to your preferred email address:

Reviewer: _main.tf_ is inside src folder. I suggested adding this reference for clarification.

}
```

> You've added a New Relic alert channel, subscribed the demo policy to the alert channel, and added a module hosted on Github.

Reviewer: Here, you added a New Relic alert channel, subscribed the demo policy to the alert channel, and added a module hosted on Github.

</Step>

<Step>

Run `terragrunt init` and then run `terragrunt apply`:

```bash
terragrunt init
terragrunt apply
```

> After Terraform finishes processing, your alert policy has two conditions in your and one alert channel.

Reviewer: ```suggestion
After Terraform finishes processing, your alert policy has two conditions and one alert channel.
```

</Step>

<Step>

## Use your environment as a Terragrunt variable

With Terragrunt, you can add the name of the environment you're running to the name of the data you're creating, making your resource more identifiable in New Relic.

In the root _terragrunt.hcl_ file, create an input for `env_name`:

```hcl
inputs = {
	env_name = "develop"
}
```

> In the _main.tf_, file add a new variable called `env_name`:

Reviewer: Again, reference to the location of _main.tf_ file can be helpful.

}
```

> Adding a `locals` block creates a local variable. Using the replace built-in function removes the unwanted parts of the relative path. And updating the `inputs` block to use the local variable.

Reviewer: Can we rephrase this? The language doesn't seems to convey the context properly. I suggest something like. 
```suggestion
Here, you added a `locals` block to create a local variable and used the built-in `replace` function to remove the unwanted parts of the relative path.  Then, you updated the `inputs` block to use the local variable.
```

Author: Agreed. I mostly copied this from above and changed the commas to periods. I should have edited the language. Good catch!

[output]
[output] Plan: 2 to add, 1 to change, 2 to destroy.
```

Your new policy name is "dev: My Demo Policy".

</Step>

<Step>

Run `terragrunt apply` to update your configurations:

```bash
terragrunt apply
```

</Step>

</Steps>

## Move your state to remote storage

At the moment, your state file is local. Now, you update your Terraform configurations to store it in Amazon S3.

<Callout variant="tip">

Since Terragrunt allows you to configure multiple environments, you should store state files in their own S3 buckets so they don't overwrite each other.

</Callout>

<Steps>

<Step>

[Create an S3 bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html) for you development state file.

</Step>

<Step>

[Create an IAM user](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html) with [permissions to store files in your bucket](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_s3_rw-bucket.html).

> In your root _terragrunt.hcl_, add a `remote_state` block that tells Terragrunt where to place your file in S3:

Reviewer: Shouldn't this be a separate step?

Author: yep
